examples/js/exporters/DracoExporter.js

```diff
@@ -0,0 +1,275 @@
+'use strict';
+
+/**
+ * Export draco compressed files from threejs geometry objects.
+ *
+ * Draco files are compressed and usually are smaller than conventional 3D file formats.
+ *
+ * The exporter receives a options object containing
+ *  - decodeSpeed, indicates how to tune the encoder regarding decode speed (0 gives better speed but worst quality)
+ *  - encodeSpeed, indicates how to tune the encoder parameters (0 gives better speed but worst quality)
+ *  - encoderMethod
+ *  - quantization, indicates the presision of each type of data stored in the draco file in the order (POSITION, NORMAL, COLOR, TEX_COORD, GENERIC)
+ *  - exportUvs
+ *  - exportNormals
+ *
+ * @class DRACOExporter
+ * @author tentone
+ */
+
+THREE.DRACOExporter = function () {};
+
+THREE.DRACOExporter.prototype = {
+
+	constructor: THREE.DRACOExporter,
+
+	parse: function ( geometry, options ) {
+
+
+		if ( DracoEncoderModule === undefined ) {
+
+			throw new Error( 'THREE.DRACOExporter: required the draco_decoder to work.' );
+
+		}
+
+		if ( options === undefined ) {
+
+			options = {
+
+				decodeSpeed: 5,
+				encodeSpeed: 5,
+				encoderMethod: THREE.DRACOExporter.MESH_EDGEBREAKER_ENCODING,
+				quantization: [ 16, 8, 8, 8, 8 ],
+				exportUvs: true,
+				exportNormals: true,
+				exportColor: false
+
+			};
+
+		}
+
+		var dracoEncoder = DracoEncoderModule();
+		var encoder = new dracoEncoder.Encoder();
+		var builder = new dracoEncoder.MeshBuilder();
+		var mesh = new dracoEncoder.Mesh();
+
+		if ( geometry.isBufferGeometry === true ) {
+
+			var vertices = geometry.getAttribute( 'position' );
+			builder.AddFloatAttributeToMesh( mesh, dracoEncoder.POSITION, vertices.count, vertices.itemSize, vertices.array );
+
+			var faces = geometry.getIndex();
+			builder.AddFacesToMesh( mesh, faces.count, faces.array );
+
+			if ( options.exportNormals === true ) {
+
+				var normals = geometry.getAttribute( 'normal' );
+
+				if ( normals !== undefined ) {
+
+					builder.AddFloatAttributeToMesh( mesh, dracoEncoder.NORMAL, normals.count, normals.itemSize, normals.array );
+
+				}
+
+			}
+
+			if ( options.exportUvs === true ) {
+
+				var uvs = geometry.getAttribute( 'uv' );
+
+				if ( uvs !== undefined ) {
+
+					builder.AddFloatAttributeToMesh( mesh, dracoEncoder.TEX_COORD, uvs.count, uvs.itemSize, uvs.array );
+
+				}
+
+			}
+
+			if ( options.exportColor === true ) {
+
+				var colors = geometry.getAttribute( 'color' );
+
+				if ( colors !== undefined ) {
+
+					builder.AddFloatAttributeToMesh( mesh, dracoEncoder.COLOR, colors.count, colors.itemSize, colors.array );
+
+				}
+
+			}
+
+		} else if ( geometry.isGeometry === true ) {
+
+			var numFaces = geometry.faces.length;
```

Comment:
Maybe it's easier to just convert an instance of `Geometry` to `BufferGeometry` and then use the same code path.

Reply:
Yeah it is a better approach makes it easier to maintain latter down the road and the code its doing pretty much the same as the built in conversion anyway.

I was trying to understand why my draco files were smaller for Geometry objects i was not exporting UV's propeties properly.

Will update the PR soon

Reply:
Updated the PR but i still need to change it to ensure that the buffer geometry is using faces, that we for the draco file.

Otherwise the encoding process will fail.

Reply:
Related #6324

Reply:
Note that unfortunately Geometry—>BufferGeometry conversion has two inefficient side effects:

- vertex count increases significantly, even after toIndexed() call
- blank vertex colors are added to each vertex

The first can be fixed with BufferGeometryUtils.mergeVertices() but that’s probably better left to userland code. I’m not suggesting any change here (perhaps a warning on Geometry?), but just something to be aware of.